src/xtgeo/well/_well_oper.py

```diff
@@ -438,3 +429,52 @@ def report_zonation_holes(self, threshold=5):
         clm = ["INDEX", "X_UTME", "Y_UTMN", "Z_TVDSS", "Zone", "Well"]
 
     return pd.DataFrame(wellreport, columns=clm)
+
+
+def mask_shoulderbeds(self, inputlogs, targetlogs, nsamples, strict):
+    """Mask targetlogs around discrete boundaries."""
+    logger.info("Mask shoulderbeds for some logs...")
+
+    # check that inputlogs exists and that they are discrete, and targetlogs
+    useinputs = []
+    for inlog in inputlogs:
+        if inlog not in self._wlogtypes.keys() and strict is True:
+            raise ValueError(f"Input log {inlog} is missing and strict=True")
+        if inlog in self._wlogtypes.keys() and self._wlogtypes[inlog] != "DISC":
+            raise ValueError(f"Input log {inlog} is not of type DISC")
+        if inlog in self._wlogtypes.keys():
+            useinputs.append(inlog)
+
+    usetargets = []
+    for target in targetlogs:
+        if target not in self._wlogtypes.keys() and strict is True:
+            raise ValueError(f"Target log {target} is missing and strict=True")
+        if target in self._wlogtypes.keys():
+            usetargets.append(target)
+
+    if not useinputs or not usetargets:
+        logger.info("Mask shoulderbeds for some logs... nothing done")
+        return False
+
+    # perform filtering
+    def _growfilter(bseries, nleft):
+        if not nleft:
+            return bseries
+        else:
```

Comment:
this `else` is not needed due to the `return` above (flake/pylint complains?)

Reply:
You're right. pylint complains; flake8 (in precommit) was quiet.